### DISCUSSION TITLE
CI/CD - Release-Zeiträume

### DISCUSSION THREAD

**Discussion by dippeal** [openjverein/jverein#80] [General]
Hallo Leute,

wann würdet ihr neue Tags und Builds erstellen?
- Monatlich/Quartalsmäßig/fester Zeitpunkt
- Nach jedem Bugfix
- Bei jedem neuen Feature
- Wenn eine Anzahl n Pull requests fertig sind
- Immer wenn es eine neue DB Update Datei gibt

Und wer soll/darf/will Tags und Builds für openjverein erstellen?

**Comment by almereyda** [comment]
Aus meiner Sicht sind

- instabile Dailies nach jedem nach `main` gemergeten Bugfix
- stabile, manuelle Releases nach einer Reihe von zusammenpassender PRs

gut. Ersteres davon könnte automatisiert geschehen.

Steigende Migrationen wären in beiden oberen Fällen enthalten und blieben so getrennt von stabilen Releases testbar.

**Reply by willuhn** [reply]
Ich handhabe es mit den Releases bei Hibiscus nach Bedarf. Entweder es hat sich eine größere Menge Neuerungen ergeben oder es sind wichtige Bugfixes enthalten. Also keine fixen Intervalle. Meist kommt dann so ca. einmal im Quartal hinreichend viel für ein neues Release zusammen.

Nightly-Builds für JVerein würde ich auch gut finden. Ich weiss aber nicht, ob Github Actions mit Ant möglich sind. Eine Gradle-Migration ist wegen der Compile-Abhängigkeit zu Jameica nicht ganz trivial. Ich weiss nicht, ob man eine Github Action so schreiben kann, dass zum Compilieren nicht nur JVerein sondern auch Jameica mit ausgecheckt wird.

**Reply by almereyda** [reply]
Ja die GitHub Actions sind eigentlich nur komplizierte Shellskripte, die so ähnlich wie Ansible Playbooks eine sehr ausführliche Syntax haben. Damit lässt sich schon viel erreichen. Ich habe leider keine Erfahrungen mit Javaprogrammen, würde aber im Rahmen von #39 versuchen eine existierende Bauanleitung, die ich zuvor lokal verifizieren würde, zu übertragen.

#40 klingt damit erstmal nach einer größeren Aufräum- und Portierungsarbeit, auf die wir erstmal nicht zwingend warten müssen. Sollte das einmal portiert werden, müssten nur auch die Automatisierungen nachfolgen. Wenn erst die Bauumgebungen vereinheitlicht würden, wäre es langfristig gesehen bequemer und nützlicher den Ablauf erst danach ins CI/CD System zu übertragen.

**Reply by dippeal** [reply]
Habe den https://github.com/openjverein/jverein/pull/83 erstellt, welcher die zwei genannten Funktionen unterstützt.

**Comment by dippeal** [comment]
@almereyda willst du dich noch als Mitglied der openjverein Organisation eintragen lassen (z.B. https://github.com/openjverein/jverein/issues/46)? Dann kannst du PR approven. Ich glaube du könntest da gut unterstützen.

**Reply by almereyda** [reply]
Ja stimmt, wenn das in einer so außenstehenden Position wie meiner Sinn macht. Gucke gerne immer mal wieder über solche Sachen rüber.

**Comment by dippeal** [comment]
Ich habe die 2.8.21 released. Es sind so viele tolle features und fixes dazu gekommen, dass ich es für nötig hielt ein offizielles release zu erstellen.
